spec/classes/plugins_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'spec_helper'
+
+describe 'sensu::plugins', :type => :class do
+  on_supported_os({facterversion: '3.8.0'}).each do |os, facts|
+    if facts[:osfamily] == 'Debian'
+      package_require = ['Class[Sensu::Repo::Community]', 'Class[Apt::Update]']
```

Comment:
What I've done in the past is outside the describe set the data such as

```ruby
platforms => {
  debian => {
    :package_require => ['Class[Sensu::Repo::Community]', 'Class[Apt::Update]'],
    :dependencies => ['make','gcc','g++','libssl-dev'],
  },
  el => {
    :package_require => ['Class[Sensu::Repo::Community]'],
    :dependencies => ['make','gcc','gcc-c++','openssl-devel'],
  },
}
```

Then keep your if statement and use that to assign the data

```
package_require = platforms[#{facts[:osfamily]}[:package_require]
```

This is useful as the data changes and we add more platforms. With the current code you assume if osfamily is not Debian then it must be RedHat. By using the above method, it makes it very clear which platform's data is being used.

Reply:
Took similar approach but defined a function in spec_helper that can be used by all the unit tests so these are all in one place.